### ISSUE TITLE
Error: Invariant Violation: addComponentAsRefTo(...)

### ISSUE THREAD

**Issue by tsingson** [facebook/react#4281]
my code

```
import React from 'react';
import ReactDOM from 'react-dom';

var MyComponent = React.createClass({
  render: function() {
    return (
      <div>
      <Zoo />
      <div>Hello World</div>
      </div>
      );
  }
});

var Zoo = React.createClass({
  render: function() {
    return (
      <div>
        Giraffes name: <input ref="giraffe" />
      </div>
    );
  },

  showName: function() {
    // Previously:
    // var input = this.refs.giraffe.getDOMNode();
    var input = this.refs.giraffe;
    alert(input.value);
  }
});

var node = document.getElementById('app');
ReactDOM.render(<MyComponent />, node);

```

the error msg:
```
Uncaught Error: Invariant Violation: addComponentAsRefTo(...): Only a ReactOwner can have refs. This usually means that you're trying to add a ref to a component that doesn't have an owner (that is, was not created inside of another component's `render` method). Try rendering this component inside of a new top-level component which will hold the ref.n @ invariant.js:39r.addComponentAsRefTo @ ReactOwner.js:67o @ ReactRef.js:23i.attachRefs @ ReactRef.js:39o @ ReactReconciler.js:21a.notifyAll @ CallbackQueue.js:65d.close @ ReactReconcileTransaction.js:80r.closeAll @ Transaction.js:202r.perform @ Transaction.js:149m @ ReactMount.js:271r.perform @ Transaction.js:136d.batchedUpdates @ ReactDefaultBatchingStrategy.js:62a @ ReactUpdates.js:94Q._renderNewRootComponent @ ReactMount.js:402o.measure.i @ ReactPerf.js:66Q._renderSubtreeIntoContainer @ ReactMount.js:469Q.render @ ReactMount.js:489o.measure.i @ ReactPerf.js:66c @ index.js:33t @ bootstrap c40e0ab928560980da40:19(anonymous function) @ bootstrap c40e0ab928560980da40:39(anonymous function) @ bootstrap c40e0ab928560980da40:39
```

anyone help for this ??

**Comment by jimfb** [comment]
Based on your code, I'm assuming you're playing with the 0.14.0-beta?

Your code looks fine to me, and it runs fine in my environment.  How are you running this code?  Can you provide a jsfiddle?

**Comment by jedwards1211** [comment]
I'm seeing this in React 0.13.3 and it makes no sense!  I'm trying to use the google-map-react component and it doesn't do anything weird with refs.

**Comment by jimfb** [comment]
We have seen this error if you have multiple instances of React loaded; so that might be an idea of something to take a look at if you are using a pre-packaged component.  Sometimes a second instance of React can get pulled in as a dependency to one of your components/whatever.

**Comment by jedwards1211** [comment]
Yeah that's exactly what was triggering it for me, but I still don't understand why, it was loading /~/react twice, same exact code.  Somehow the PrefetchPlugin fixed it for me, but I don't really know why.

**Comment by spicyj** [comment]
You probably have two copies of react in your repo. `npm ls` or `npm ls | grep react` might help illuminate.

**Comment by jedwards1211** [comment]
I think in my case it might have been one piece of code requiring `react` and another requiring `react/addons`.  I should probably just webpack alias `react` to the latter.

**Comment by spicyj** [comment]
That alone shouldn't cause problems.

**Comment by chandu0101** [comment]
I am also getting this error in my app

```js

 npm ls | grep react
├─┬ react@0.14.0-beta2

index.js 
var React = require('react'),
     ReactDOM = require('react-dom');

in render method i am rendering a simple div

React.createElement("div",json(ref = "heh"),"huhuh")
```

**Comment by sanjay1909** [comment]
I too had the similar issue,  I checked for two copies of React and ref , 

> but no success

Then i tried this small test, i created two library project(x and y ) and one application project
**First Test**
1. I made React and React-Dom as peerDependencies in both the library projects,
2. Project  - Y depends on Project X
3. And then application project depends on React, React-Dom and Project-Y

> No error was reported.

**Second Test**

1. In  project X i used the "refs" and i made project Y use that component in project X which had "refs".
2. In application project I used the project Y component.

> This time error was reported. though both the library project had React, React-Dom as external and also resided in peerDependencies. 

It turns out that the library project i was using has **main**  property in package.json was pointing to **distributable** version path, rather pointing to **npm module** version path, 

That caused all the trouble ..... Thought of sharing with you guys. 

**Double check the library package.json too**
